src/driver/amdxdna/aie2_msg_priv.h

```diff
@@ -387,6 +389,53 @@ struct async_event_msg_resp {
 	enum async_event_type	type;
 } __packed;
 
+/***************Start of event tracing data struct ****************/
+#define TRACE_EVENT_BUFFER_SIZE 0x100000
+#define EVENT_TRACE_CATEGORY_OVERVIEW (0x00000001u)
+#define EVENT_TRACE_CATEGORY_TCT      (0x00000002u)
+#define EVENT_TRACE_CATEGORY_LATENCY  (0x00000004u)
+#define EVENT_TRACE_CATEGORY_LATENCY2 (0x00000008u)
+
+enum event_trace_destination {
+    EVENT_TRACE_DEST_DEBUG_BUS,
+    EVENT_TRACE_DEST_DRAM,
+    EVENT_TRACE_DEST_COUNT
+};
+
+enum event_trace_timestamp {
+    EVENT_TRACE_TIMESTAMP_FW_CHRONO,
+    EVENT_TRACE_TIMESTAMP_CPU_CCOUNT,
+    EVENT_TRACE_TIMESTAMP_COUNT
+};
+
+struct start_event_trace_req {
+    uint32_t event_trace_categories;
+    enum event_trace_destination event_trace_dest;
+    enum event_trace_timestamp event_trace_timestamp;
+	//DRAM log buffer address and size
+    uint64_t dram_buffer_address;
+    uint32_t dram_buffer_size;
+};
+
+struct start_event_trace_resp {
+    enum aie2_msg_status status;
+    uint32_t msi_idx;
+    uint64_t current_timestamp;
+};
+
+struct stop_event_trace_resp {
+    enum aie2_msg_status status;
+};
+
+struct set_event_trace_categories_req {
+     uint32_t event_trace_categories;
+};
+
+struct set_event_trace_categories_resp {
+    enum aie2_msg_status status;
+};
```

Comment:
For all *_req and *_resp struct, "__packed" is needed.

BTW, I don't see "stop_event_trace_req". Did I miss something?

Reply:
@mamin506  I think it's still a Draft PR, Vinit will be posting more updates as he keeps working on it.

Reply:
Right, no need to review until the PR is migrated to a formal one, which means that:
1. The unit test has passed
2. CI passed (no regression and checkpatch clean)

Reply:
Hi team, thanks for review initiation.
Yes, the code is still in daft phase, pushed code for review in case of debug support with global team.
Will add comments if any info required.

Reply:
> For all *_req and *_resp struct, "__packed" is needed.
> _Thanks, updated and resp issue is resolved due to size mismatch at FW. _

> BTW, I don't see "stop_event_trace_req". Did I miss something?
> _Will add further._